### PR TITLE
Add 9-Slice support to UiImage

### PR DESCRIPTION
**Pull request by mrhatman**
## Description

Added 9-Slice to UiImage. Including prefab support and updated the ui example to include a 9-Slice. 

## Additions

-  Add `NineSlice` support to `UiImage`
-  Add `NineSlice` support to `UiImageLoadPrefab`

## PR Checklist

By placing an x in the boxes I certify that I have:

- N/A Updated the content of the book if this PR would make the book outdated.
- [x ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new code added in this PR.
- [x ] Acknowledged that by making this pull request I release this code **and image** under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [x] Ran `cargo clippy --all --features "empty"`
- [x] Ran `cargo test --all --features "empty"`

### CODE REVIEW

File: amethyst_ui/src/prefab.rs [hunk 1]
```diff
@@ -343,6 +343,19 @@ pub enum UiImageLoadPrefab {
     PartialTexture(TexturePrefab, f32, f32, f32, f32),
     /// Solid color image
     SolidColor(f32, f32, f32, f32),
+    /// 9-Slice image
+    NineSlice {
+        x_start: i32,
```
Comment (azriel91) [inline amethyst_ui/src/prefab.rs]: Should likely use `u32`s for all of these

File: examples/assets/ui/example.ron [hunk 2]
```diff
@@ -61,6 +61,31 @@ Container(
             ),
             image: PartialTexture(File("texture/test_texture.png", ("IMAGE", ())),0.34,0.0,1.0,0.333),
         ),
+        //9-Slice
+        Image(
+            transform: (
+                id: "Partial Texture Logo",
```
Comment (azriel91) [inline examples/assets/ui/example.ron]: `"9 Slice"` or something

File: examples/assets/ui/example.ron [hunk 3]
```diff
@@ -61,6 +61,31 @@ Container(
...
+                x: -250.,
+                y: -40.,
+                width: 150.,
+                height:  80.,
```
Comment (azriel91) [inline examples/assets/ui/example.ron]: nit: double space

File: examples/assets/ui/example.ron [hunk 4]
```diff
@@ -61,6 +61,31 @@ Container(
...
+                tab_order: 1,
+                anchor: TopRight,
+                mouse_reactive: true,
+            ),
+            image: NineSlice(
+                    x_start:0,
```
Comment (azriel91) [inline examples/assets/ui/example.ron]: nit: could you use 4 space indent and add a space after each `:`

File: amethyst_ui/src/image.rs [hunk 5]
```diff
@@ -12,6 +12,29 @@ pub enum UiImage {
     PartialTexture(Handle<Texture>, TextureCoordinates),
     /// An image backed by a Sprite
     Sprite(SpriteRender),
+    /// An Image backed by a 9-sliced texture
+    NineSlice {
+        ///x starting position on the texture
```
Comment (azriel91) [inline amethyst_ui/src/image.rs]: nit: could you add a space after each `///`, and capitalize the first letter for sentences?

File: amethyst_ui/src/prefab.rs (hunk 1 above)
Comment (mrhatman) [reply amethyst_ui/src/prefab.rs]: you are correct. Fixed this here and in the UiImage definition.

File: examples/assets/ui/example.ron (hunk 2 above)
Comment (mrhatman) [reply examples/assets/ui/example.ron]: Opps. Copy Paste Error. Fixed.

File: examples/assets/ui/example.ron (hunk 3 above)
Comment (mrhatman) [reply examples/assets/ui/example.ron]: Fixed

File: examples/assets/ui/example.ron (hunk 4 above)
Comment (mrhatman) [reply examples/assets/ui/example.ron]: I believe I fixed this

File: amethyst_ui/src/image.rs (hunk 5 above)
Comment (mrhatman) [reply amethyst_ui/src/image.rs]: Fixed

**Comment by azriel91** [comment]
bors r=azriel91

**Pull request merged**